docs/CODE_CONVENTIONS.md

```diff
@@ -0,0 +1,138 @@
+    # was created. 
+    @property
+    def protected_field(self) -> SomeType:
+        return self._protected_field
+
+    # Setter should NOT be crated if field is not changed ounside this class
+    @property.setter
+    def protected_field(self, some_value: SomeType) -> None: 
+        self._protected_field = some_value
+
+    def some_method(self) -> None:
+        # Field inside the class is reffered by it's name directly. 
+        # Not by property
+        self._protected_field = SomeType()
+
+# This is a model class.
+# Getter and setter properties should NOT be used for fields of this class
+# sinse it just stores the data.
+class ModelClass:
+    def __init__(self):
+        self.data_field_1 = DataType()
+        self.data_field_2 = DataType()
+        self.data_field_3 = DataType()
+
+```
+
+
+## Comments
+
+You should follow pep8 [Comments](https://www.python.org/dev/peps/pep-0008/#comments) conventions.
+
+## Docstring conventions
+
+You should follow [Docstring conventions](https://www.python.org/dev/peps/pep-0257/).
+
+## Commit messages
+
+You should follow: [Conventional Commits 1.0.0](https://www.conventionalcommits.org/en/v1.0.0/)
+for every commit message.
+
+## Type hints
+
+You should use [Type hints](https://docs.python.org/3/library/typing.html) in code.
+
+Additional notes:
+- For every method should be defined it return type
+- If method returns nothing - return type should be specified as `None`
+  - This statement is true for every method, including method `__init__()`   
+- Type should be specified for every method parameter
+- Type hints should NOT be used for variables in code
+  - An exception of this rule: type hints should be specified for **class fields** which type is not obvious during initialization (example: different containers)
+- In case if method can return objects with different type - `Any` return type should be specified
+- To avoid unnecessary imports - several type names could be written inside the quotes
+
+Example:
+``` python
+class ExampleClass:
+    # Method condtructor returns nothing
+    def __init__(self) -> None:
+        # Field type IS NOT hinted because it is obvious from initialization
+        self._field_1 = SomeType1()
+        # Field type IS NOT hinted because it could be easily
+        # inspected by viewing create_field_method specification
+        self._field_2 = create_field_method()
+        # Field type IS hinted because it is no way to
+        # understand which elements should contain this list - during initialization
+        self._field_3: List[SomeType3] = []
+    
+    # All method arguments marked with type hints
+    # Argument2Type is written inside quotes to avoid unnecessary import of Argument2Type in current module
+    def some_method_1(self, argument_1: Argument1Type, argument_2: 'Argument2Type') -> ReturnValueType:
+        # Local variable type should not be hinted
+        local_varuable = LocalVariable() 
```

Comment:
Mistake. "local_variable" name should be

Reply:
Done